modules/gis/src/lib/geojson-to-binary.js

```diff
@@ -73,87 +68,133 @@ function firstPass(features) {
 
           // eslint-disable-next-line max-depth
           for (const coord of flatten(polygon)) {
-            // eslint-disable-next-line max-depth
-            if (coord.length === 3) maxCoordLength = 3;
+            coordLengths.add(coord.length);
           }
         }
         break;
       default:
         throw new Error(`Unsupported geometry type: ${geometry.type}`);
     }
+
+    if (feature.properties) {
+      for (const key in feature.properties) {
+        const val = feature.properties[key];
+        numericProps[key] = numericProps[key] ? isNumeric(val) : numericProps[key];
```

Comment:
The logic may well be correct, but I get confused here. Do we want this to be true set only if all the values are numeric? Maybe the following is easier to follow?

`
// initalize
for (const key in feature.properties) {propIsNumericKey[key] = true;}
...
// each iteration, check if still considered numeric
if (propIsNumericKey[key]) {
  propIsNumericKey[key] = Number.isFinite(val);
}

Reply:
> Do we want this to be true only if all the values in all features are numeric? 

I think so, yes.

> If so maybe the following is easier to follow?

That snippet is initializing based on the properties in the first feature? The problem with that is that not all features are required to have all keys. With MVT in particular, if I'm not mistaken, a property could be defined in only a single feature within a layer.

So if we want `numericKey` to be `true` if the values for that prop in all features are numeric **and** exist, then initializing based on the properties in the first feature are fine. Otherwise, to initialize based on all properties I think it would be necessary to loop over all features just to do the initialization.

I'm currently interpreting `numericKey` should be `true` if the values for that prop in all features are either numeric **or** do not exist. So I need to keep track of 1) have I seen this key in a previous feature, 2) has it been non-numeric in any previous feature.

It might be good to add a comment like
```js
// If property has been numeric in all previous features in which the
// property existed, check if numeric in this feature
```

Reply:
Sounds good. This can get pretty sophisticated, so you should draw the line here for now and then we can come back and make further improvements if we feel so inclined.

For later:
- An approach used in binary columnar tables is to optionally generate a second `null` array to track which objects are not present (e.g. a `Uint8Array`).
- We could detect the type of the array:
   -  For each element check if integer. If row is all integer, generate `Uint32Array`
   - Track max/min values of all values
    - if (-32000, 32000), generate `Uint16Array`
    - If (-128, 127) generate `Uint8Array`
    - etc

Maybe create a follow up tasks with such ideas for a later date?

Reply:
> This can get pretty sophisticated, so you should draw the line here for now and then we can come back and make further improvements if we feel so inclined.

Good idea, it's clear this can be a black hole.

One more idea I thought of that can be a future task: should `numericKeys` depend on the geometry type?

For example, if I have a set of GeoJSON features where a numeric property `x` only exists on `Point` features, and I have 10 `Point` features but 1000 `Polygon` features, the current code instantiates a `Float32Array` for _every Polygon vertex_, when every vertex is missing data. 

I don't think this would necessarily be uncommon either; I have a `mountain_peak` layer of points in my MVTs with a numeric `elevation` value, but no other layer has that property, so there would be `# line positions + # polygon positions` extra empty data just for that property.

Reply:
Yes that is a good observation, that can be another optimization.

Right now I think we should be concerned with the "combinatorial explosion" of options + input dependent outputs, specifically relating to the testability and ease-of-use of it all. 

Less optionality is probably good for now

Reply:
Yes, you're right. It's easy to go on a feature-adding spree. If there's anything you think should be taken out let me know.

Now I'm pivoting to adding more tests... My initial approach would be to add tests to `firstPass` and `secondPass` separately, to more easily test edge cases. Do you know if it's possible to write tests for a non-`exported` function in JS? Or is it ok to export the functions for use in tests? I saw one approach on the web like
```js
export const _testing = [firstPass, secondPass]
```

Reply:
Yes I sometimes add a `TEST_EXPORTS`: `const TEST_EXPORTS = {firstPass, secondPass}`

Also by using an object you can then do:
```
import {TEST_EXPORTS} from ...; 
const {firstPass, secondPass} = TEST_EXPORTS;
```